specs/tendrl_ansible_grafana_installation.adoc

```diff
@@ -0,0 +1,156 @@
+= Tendrl ansible to install grafana
+
+Needs ansible playbook scripts for automatic installation and configuration of
+grafana and tendrl metrics in the tendrl server.
+
+
+== Problem description
+
+Grafana will be used in tendrl to trigger alerts and visualize the tendrl
+metrics. Currently, Tendrl-ansible is used to install and configure tendrl
+packages. It should also support installation and configuration of tendrl
+metrics and grafana package installation.
+This specification provides the required changes in tendrl-ansible tool to
+support installation and configuration of tendrl metrics and grafana.
+
+
+== Use Cases
+
+* Provide playbook for tendrl-ansible to install tendrl-metrics and grafana
+along with server installation.
+
+
+== Proposed change
+
+* Add the repo to install latest grafana and its dependencies
+
+* Install tendrl metrics, grafana, fontconfig, freetype* and urw-fonts
+
+* Configure tendrl metrics and  grafana
+  This should add the required environment variables, set log path, database
+  connection details, data source and the node details
+
+* Enable the tendrl metrics and grafana server to start at boot time
+
+* Start the grafana-server service
+
+* Document the ansible playbook configuration, tendrl metrics activation
+  details in wiki
+
+
+=== Alternatives
+
+None
+
+=== Data model impact
+
+None
+
+=== Impacted Modules:
+
+==== Tendrl Ansible impact:
+
+* Update roles in site.yml.sample about the repo details install grafana
+and its dependencies for tendrl-server host installation.
+- hosts: tendrl-server
+  ...
+  roles:
+    - tendrl-copr
+    - grafana
+    ...
+
+* Add repo details under tendrl-ansible/roles
+  files directory with keys and the repo files
+  tasks to install the repos
+
+* Create and add tasks to install grafana and tendrl metrics packages
+  in tendrl-ansible/roles/tasks
+
+* Include tasks to configure grafana and tendrl metrics packages
+  in tendrl-ansible/roles/tasks
+
+* Update tendrl-ansible/roles/tendrl-server/handlers/main.yml
+  to check the status of grafana server service and start
```

Comment:
Sure, the sample playbook could be as bellow:
- name: Install Grafana package
  yum:
    name: "{{ grafana_pkg }}"
    state: present
  notify:
    - Restart grafana

- name: Configure Grafana
  template:
    src: grafana.ini.tendrl
    dest: /etc/grafana/grafana.ini
  notify:
    - Restart grafana
  tags:
    - grafana_config

- name: Restart grafana
  service:
    name: grafana-server
    state: restarted

- name: Make sure grafana service is enabled and running
  service:
    name: grafana-server
    enabled: yes
    state: running